src/Gemini.Modules.CodeCompiler/CodeCompiler.cs

```diff
@@ -52,6 +52,31 @@ public Assembly Compile(IEnumerable<SyntaxTree> syntaxTrees, IEnumerable<Metadat
                 }
                 _output.AppendLine("------ Compile finished");
                 ms.Seek(0, SeekOrigin.Begin);
+                
+                if (exportDll)
+                {
+                    if (exportDir == null)
+                    {
+                        exportDir = typeof(Gemini.AppBootstrapper).Assembly.Location;
+                    }
+                    if (!outputName.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
+                        outputName += ".dll";
+                    _output.AppendLine("------ Exporting to DLL");
+                    var dllPath = System.IO.Path.Combine(exportDir, outputName);
+                    try
+                    {
+                        if (File.Exists(dllPath))
+                            File.Delete(dllPath);
+                        ms.WriteTo(new FileStream(dllPath, FileMode.Create));
+                        _output.AppendLine("------ Export finished");
+                        _output.AppendLine("-> " + dllPath);
+                    }
+                    catch (Exception ex)
+                    {
+                        _output.AppendLine("------ Export failed");
+                        _output.AppendLine(ex.Message);
+                    }
+                }
```

Comment:
I would think that it should still have it load into memory. I added the parameters into the existing method to prevent duplicated code.